smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/providers/wiring/Wiring.java

```diff
@@ -504,7 +515,13 @@ public void materialize(ChannelRegistry registry) {
         private <T extends MessagePublisherProvider<?>> void registerEmitter(ChannelRegistry registry, int def) {
             EmitterFactory<?> emitterFactory = getEmitterFactory(configuration.emitterType());
             T emitter = (T) emitterFactory.createEmitter(configuration, def);
-            Publisher<? extends Message<?>> publisher = emitter.getPublisher();
+            Multi<? extends Message<?>> publisher = Multi.createFrom().publisher(emitter.getPublisher());
+            for (PublisherDecorator decorator : getSortedInstances(decorators)) {
+                // TODO do not apply ContextDecorator here, it messes with custom emitter transactions
+                if (!(decorator instanceof ContextDecorator)) {
+                    publisher = decorator.decorate(publisher, configuration.name(), false);
```

Comment:
So, you consider the emitter as a "not connector", which is exact. However, I was wondering if we need to be specific in this case and introduce a second flag (I'm not a big fan of signatures containing two exclusive flags, having an enum would be better).

However, do we need to know when it's an emitter?